Buildings/Experimental/OpenBuildingControl/CDL/SetPoints/CalendarTime.mo

```diff
@@ -0,0 +1,360 @@
+    if unixTimeStamp < timeStampsNewYear[i] and (if i == 1 then true else unixTimeStamp >= timeStampsNewYear[i-1]) then
+      yearIndex :=i - 1;
+      year :=firstYear + i - 2;
+    end if;
+  end for;
+
+  // iterate to find the month at initialization
+  epochLastMonth := timeStampsNewYear[yearIndex];
+  month:=13;
+  for i in 1:12 loop
+    if (unixTimeStamp-epochLastMonth)/3600/24 <
+      (if i==2 and isLeapYear[yearIndex] then 1 + dayInMonth[i] else dayInMonth[i]) then
+      // construction below avoids the need of a break, which bugs out JModelica
+      month :=min(i,month);
+    else
+      epochLastMonth :=epochLastMonth + (if i == 2 and isLeapYear[yearIndex]
+         then 1 + dayInMonth[i] else dayInMonth[i])*3600*24;
+    end if;
+  end for;
+
+equation
+  // compute unix time step based on found offset
+  unixTimeStamp = time + offset + timOff;
+
+  // update the year when passing the epoch time stamp of the next year
+  when unixTimeStamp >= timeStampsNewYear[pre(yearIndex)+1] then
+    yearIndex=pre(yearIndex)+1;
+    assert(yearIndex<=size(timeStampsNewYear,1),
+      "Index out of range for epoch vector: timeStampsNewYear needs to be extended beyond the year "
+        + String(firstYear+size(timeStampsNewYear,1)));
+    year = pre(year) + 1;
+  end when;
+
+  // update the month when passing the last day of the current month
+  when unixTimeStamp >= pre(epochLastMonth) +
+      (if pre(month)==2 and isLeapYear[yearIndex]
+        then 1 + dayInMonth[pre(month)] else dayInMonth[pre(month)])*3600*24 then
+    month = if pre(month) == 12 then 1 else pre(month) + 1;
+    epochLastMonth = pre(epochLastMonth) +
+      (if pre(month)==2 and isLeapYear[yearIndex]
+        then 1 + dayInMonth[pre(month)] else dayInMonth[pre(month)])*3600*24;
+  end when;
+
+  // compute other variables that can be computed without using when() statements
+  hourSampleTrigger =sample(hourSampleStart, 3600);
+  when hourSampleTrigger then
+    if pre(firstHourSampling) then
+      hour = integer(floor(rem(unixTimeStamp,3600*24)/3600));
+    else
+      hour = if (pre(hour) == 23) then 0 else (pre(hour) + 1);
+    end if;
+    firstHourSampling = false;
+  end when;
+
+  daySampleTrigger =sample(daySampleStart, 86400);
+  when daySampleTrigger then
+    if pre(firstDaySampling) then
+      daysSinceEpoch = integer(floor(unixTimeStamp/3600/24));
+      weekDay=integer(rem(4+daysSinceEpoch-1,7)+1);
+
+    else
+      daysSinceEpoch = pre(daysSinceEpoch) + 1;
+      weekDay = if (pre(weekDay) == 7) then 1 else (pre(weekDay) + 1);
+    end if;
+    day = integer(1+floor((unixTimeStamp-epochLastMonth)/3600/24));
+
+    firstDaySampling = false;
+  end when;
+
+  // using Real variables and operations for minutes since otherwise too many events are generated
+  minute = (unixTimeStamp/60-daysSinceEpoch*60*24-hour*60);
+
+  annotation (
+    defaultComponentName="calTim",
+  Documentation(revisions="<html>
+<ul>
+<li>
```

Comment:
I usually close Dymola if I need to make sure that all files are saved. Dymola also has a function to save all files, but this, at least in the past, lead to small formatting changes in many files which made version control difficult.